assignments/lambda-classes.js

```diff
@@ -1 +1,146 @@
 // CODE here for your Lambda Classes
+class Person {
+    constructor(personAttributes){
+        this.name = personAttributes.name;
+        this.age = personAttributes.age;
+        this.location = personAttributes.location;
+        this.gender = personAttributes.gender;
+    }
+    speak() {
+        console.log(`Hello my name is ${this.name}, I am from ${this.location}`);
+    }
+}
+
+class Instructor extends Person {
+    constructor(instructorAttributes){
+        super(instructorAttributes);
+        this.specialty = instructorAttributes.specialty;
+        this.favLanguage = instructorAttributes.favLanguage;
+        this.catchPhrase = instructorAttributes.catchPhrase;
+    }
+    demo(subject){
+        console.log(`Today we are learning about ${subject}`);
+    }
+    grade(student, subject){
+        console.log(`${student.name} receives a perfect score on ${subject}`);
+    }
+    updateGrade(student){
+        let gradeChange = Math.floor(Math.random() * (100 - (-100)) + -100);
+        let newGrade = student.grade + gradeChange
+        if (newGrade < 0) {
+            student.grade = 0;
+        } else if (newGrade > 100) {
+            student.grade = 100;
+        } else {
+            student.grade += gradeChange;
+        }
+    }
+}
+
+class Student extends Person {
+    constructor(personAttributes){
+        super(personAttributes);
+        this.previousBackground = personAttributes.previousBackground;
+        this.className = personAttributes.className;
+        this.favSubjects = personAttributes.favSubjects;
+        this.grade = personAttributes.grade;
+    }
+    listsSubjects() {
+        this.favSubjects.forEach(element => console.log(element));
+    }
+    PRAssignment(subject){
+        console.log(`${this.name} has submitted a PR for ${subject}`);
+    }
+    sprintChallenge(subject){
+        console.log(`${this.name} has begun sprint challenge on ${subject}`);
+    }
+    graduate() {
+        if (this.grade > 70) {
+            console.log(`Huzzah! ${this.name} can graduate!`);
+        } else {
+            console.log(`${this.name} isn't quite ready to graduate yet. Let's go back to grading his/her work`);
+        }
+    }
+}
+
+class ProjectManager extends Instructor {
+    constructor(pmAttributes) {
+        super(pmAttributes);
+        this.gradClassName = pmAttributes.gradClassName;
+        this.favInstructor = pmAttributes.favInstructor;
+    }
+    standUp(channel){
```

Comment:
For future reference, you will want to return from these methods, and clean up console.logs since, really console.logs are only really used in a developing sense and will not return values to be used in the DOM in React.  It is just a great way to record errors and a quick way to check vars, but it only really "returns" undefined - and we can discuss the difference between the return of a function and what console.log is doing under the hood after the sprint, if you want.

Reply:
Arrow functions which contain one line expressions after the arrow are the only time you will be able to drop the reserved word `return` as it is implicitly returned there... I know this isn't an example of an arrow function but I just wanted to mention that since it was contextually relevant.